src/librustc_typeck/check/op.rs

```diff
@@ -415,6 +415,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         (lhs_ty, rhs_ty, return_ty)
     }
 
+    fn add_type_neq_err_label(
+        &self,
+        err: &mut errors::DiagnosticBuilder<'_>,
+        span: Span,
+        ty: Ty<'tcx>,
+    ) {
+        err.span_label(span, ty.to_string());
+        if let FnDef(..) = ty.sty {
+            err.span_label(span, "did you forget `()`?");
+        }
```

Comment:
Could you add a comparison between the return type of this uncalled function and the other ty in order to have a higher degree of confidence that the suggestion would be correct?

Also could you use `span_suggestion` instead of `span_label` pointing at the same span, but with a suggested text of the current span's snippet + "()"? This should be done only if the function definition has no arguments, if it has multiple arguments it should suggest something like "( /* arguments */ )" instead.

Reply:
Thank you so much.
I improved the error message.